private/path/fabridquery/matchlist.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 ETH Zurich
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fabridquery
+
+import (
+	"fmt"
+
+	"github.com/scionproto/scion/pkg/experimental/fabrid"
+)
+
+type MatchList struct {
+	SelectedPolicies []*Policy
+}
+
+// Copy creates a deep copy of the MatchList object, including the list of selected policies.
```

Comment:
Are you sure that this makes a deep copy? For me this looks like it just copies the pointer array but not the policies itself.

Reply:
Removed "deep"